### PR TITLE
Allow overriding the Lmod GPU driver check

### PR DESCRIPTION
**Pull request by ocaisa**

### CODE REVIEW

**Comment by ocaisa** [comment]
@TopRichard This should solve the issue you saw

**Comment by ocaisa** [comment]
@TopRichard It turns out we already had a workaround for this in our scripts, so you must have been running something else?

https://github.com/EESSI/software-layer/blob/2023.06-software.eessi.io/eessi_container.sh#L466-L471

The feature is useful, so I leave the PR open

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512

File: eessi_container.sh [hunk 1]
```diff
@@ -464,10 +464,9 @@ if [[ ${SETUP_NVIDIA} -eq 1 ]]; then
         BIND_PATHS="${BIND_PATHS},${EESSI_VAR_LOG}:/var/log,${EESSI_USR_LOCAL_CUDA}:/usr/local/cuda"
         [[ ${VERBOSE} -eq 1 ]] && echo "BIND_PATHS=${BIND_PATHS}"
         if [[ "${NVIDIA_MODE}" == "install" ]] ; then
-            # We need to "trick" our LMOD_RC file to allow us to load CUDA modules even without a CUDA driver
-            # (this works because we build within a container and the LMOD_RC recognises that)
-            touch ${EESSI_TMPDIR}/libcuda.so
-            export SINGULARITY_CONTAINLIBS="${EESSI_TMPDIR}/libcuda.so"
+            # No GPU so we need to "trick" Lmod to allow us to load CUDA modules even without a CUDA driver
+            # (this variable means EESSI_OVERRIDE_GPU_CHECK=1 will be set inside the container)
+            export SINGULARITYENV_EESSI_OVERRIDE_GPU_CHECK=1
```
[the remark this answers is not in the document]
Comment (ocaisa) [reply eessi_container.sh:469]: Should be fixed in https://github.com/EESSI/software-layer/pull/579/commits/bee4b001322a0d22c79642b5b1427f2c397a9f3b

**Comment by trz42** [comment]
Because the last commit updated `scripts/utils.sh` we need to rebuild...

bot: build repo:eessi.io-2023.06-software arch:x86_64/generic

**Comment by trz42** [comment]
For completeness, we built everything once more (strictly speaking not needed, but all files will have similar timestamps)...

bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512

**Review by trz42 — APPROVED** [review]
Good work! All ingested. Let's try it out in #586.

**Pull request merged**